app/forms/idv/finance_form.rb

```diff
@@ -4,49 +4,70 @@ class FinanceForm
 
     FINANCE_TYPES = [:ccn, :mortgage, :home_equity_line, :auto_loan].freeze
 
-    attr_reader :idv_params, :finance_type, :finance_account
+    FINANCE_HTML_OPTIONS = {
+      ccn: { maxlength: 8 }
+    }
 
-    validates :finance_type, :finance_account, presence: true
+    attr_reader :idv_params, :finance_type, *FINANCE_TYPES
 
-    validates :finance_account,
+    validates :finance_type, presence: true
+
+    FINANCE_TYPES.each do |type|
+      validates type, presence: true, if: ->(form) { form.finance_type == type }
+    end
+
+    validates :ccn,
               format: { with: /\A\d{8}\z/, message: I18n.t('idv.errors.invalid_ccn') },
               if: :ccn?
 
-    validate :finance_type_valid
+    validate :validate_finance_type
```

Comment:
I think it makes sense for this to be a non-boolean since, as you said, it adds error when the finance type is invalid (and I'd expect a `?` method to return `true` or `false)

Keeping as-is is totally fine, was just curious on your thinking. 

I like the way that [email validator](https://github.com/balexand/email_validator) does `validates :my_email_attribute, email: true` and think it would be neat if we did `validates :finance_type, finance_type: true` but I think that would require an external custom validator so probably not worth it.